docs/source/tutorials/chat.rst

```diff
@@ -0,0 +1,164 @@
+reserved token IDs. This means that they will index to their own individual vectors in
+the model's learnt embedding table. The rest of the prompt template tags, :code:`[INST]`
+and :code:`<<SYS>>` are tokenized as normal text and not their own IDs.
+
+.. code-block:: python
+
+    print(tokenizer.decode(518))
+    # '['
+    print(tokenizer.decode(25580))
+    # 'INST'
+    print(tokenizer.decode(29962))
+    # ']'
+
+It's important to note that you should not place the special reserved tokens in your
+input prompts manually, as it will be treated as normal text and not as a special
+token.
+
+.. code-block:: python
+
+    print(tokenizer.encode("<s>", add_bos=False, add_eos=False))
+    # [529, 29879, 29958]
+
+When should I use a prompt template?
+------------------------------------
+
+When to use or not to use a prompt template is governed by what your desired inference
+behavior is. You should use a prompt template in the following scenarios:
+
+1. You are running inference on the base model and it was pre-trained with a prompt
+template
+2. You want to prime a fine-tuned model to expect a certain prompt structure on inference
+for a specific task
+
+It is not strictly necessary to fine-tune with a prompt template, but generally you
+want the model to perform some sort of task, which will require some formatting of
+the prompt.
+
+For example, the :class:`~torchtune.data.SummarizeTemplate` provides a lightweight
+structure to prime your fine-tuned model for prompts asking to summarize text.
+
+.. code-block:: python
+
+    f"Summarize this dialogue:\n{dialogue}\n---\nSummary:\n"
+
+You can fine-tune LLaMA2 with this template even though the model was originally pre-trained
+with the :class:`~torchtune.data.Llama2ChatFormat`, as long as this is what the model
+sees during inference. The model should be robust enough to adapt to a new template.
+
+Special tokens in LLaMA3
+------------------------
+
+LLaMA3 `overhauled <https://llama.meta.com/docs/model-cards-and-prompt-formats/meta-llama-3>`_
+the special tokens and prompt templating from LLaMA2 to better support multiturn conversations.
+
+.. code-block:: text
+
+    <|begin_of_text|><|start_header_id|>system<|end_header_id|>
+
+    {{ system_prompt }}<|eot_id|><|start_header_id|>user<|end_header_id|>
+
+    {{ user_message_1 }}<|eot_id|><|start_header_id|>assistant<|end_header_id|>
+
+    {{ model_answer_1 }}<|eot_id|><|start_header_id|>user<|end_header_id|>
+
+Now, all the additional tags are special tokens, and no prompt template is required.
+You can see how they get encoded as their own token IDs below:
+
+.. code-block:: python
+
+    from torchtune.modules.tokenizers import TikTokenTokenizer
+
+    tokenizer = TikTokenTokenizer("/tmp/Meta-Llama-3-8B/original/tokenizer.model")
+    print(tokenizer._encode_special_token("<|begin_of_text|>"))
+    # 128000
+    print(tokenizer._encode_special_token("<|eot_id|>"))
+    # 128009
```

Comment:
Could be worth a quick digression into the configurations for TikToken encoding for allowed and disallowed special tokens and how we will treat any special tokens appearing in the text just as regular text (idk if this is too in the weeds though)

Reply:
I think that's way too deep imo, and no one should be playing with those APIs